snuba/datasets/entities/outcomes_raw.py

```diff
@@ -0,0 +1,87 @@
+from datetime import timedelta
+from typing import Any, Mapping, Sequence, Tuple
+
+from snuba.datasets.entity import Entity
+from snuba.datasets.plans.single_storage import SingleStorageQueryPlanBuilder
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage
+from snuba.query.extensions import QueryExtension
+from snuba.query.logical import Query
+from snuba.query.parsing import ParsingContext
+from snuba.query.organization_extension import OrganizationExtension
+from snuba.query.processors import QueryProcessor
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+from snuba.query.processors.timeseries_column_processor import TimeSeriesColumnProcessor
+from snuba.query.timeseries_extension import TimeSeriesExtension
+from snuba.util import parse_datetime, qualified_column
+
+
+class OutcomesRawEntity(Entity):
+    def __init__(self) -> None:
+        storage = get_storage(StorageKey.OUTCOMES_RAW)
+
+        self.__time_group_columns = {"time": "timestamp"}
+        self.__time_parse_columns = ("timestamp",)
+        super().__init__(
+            storages=[storage],
+            query_plan_builder=SingleStorageQueryPlanBuilder(storage=storage),
+            abstract_column_set=storage.get_schema().get_columns(),
+            writable_storage=None,
+        )
+
+    def get_extensions(self) -> Mapping[str, QueryExtension]:
+        return {
+            "timeseries": TimeSeriesExtension(
+                default_granularity=3600,
+                default_window=timedelta(days=7),
+                timestamp_column="timestamp",
+            ),
+            "organization": OrganizationExtension(),
+        }
+
+    def get_prewhere_keys(self) -> Sequence[str]:
+        return ["project_id", "org_id"]
+
+    def get_query_processors(self) -> Sequence[QueryProcessor]:
+        return [
+            BasicFunctionsProcessor(),
+            TimeSeriesColumnProcessor(self.__time_group_columns),
+        ]
+
+    # TODO: Copied from TimeSeriesDataset. Will be refactored in a future update.
+    def time_expr(self, column_name: str, granularity: int, table_alias: str) -> str:
```

Comment:
Can we move this to `Entity`?

Reply:
No, because not all entities use these. But also I removed these functions in favour of processors later.